common: add sim-on-hardware information (#4241)
ArduPilot/ardupilot_wiki: common/source/docs/common-sim-on-hardware.rst

> .. _common-sim-on-hardware:

Reviewer: this should be a dev topic I would think...not a vehicle common topic

======================
Simulation on Hardware
======================

Simulation on Hardware allows:

-  checking that mission structure is correct
-  control surfaces move appropriately throughout a simulated mission
-  ensuring that physical failsafes (e.g. parachutes) function appropriately without putting a vehicle at risk
-  communications infrastructure works in the face of real traffic

.. note::

>    Simulation on Hardware is being improved all the time.  Do not be surprised if it is suggested you run the master branch to be able to take advantage of new features that have been added.

Reviewer: ```suggestion
   Simulation on Hardware is being improved all the time.  It is suggested you run the master branch ("latest" firmware) to be able to take advantage of any new features that have been added.
```

-----------
Limitations
-----------

> -  actuator output is not supported without recompilation.  This is currently considered a safety feature

Reviewer: ```suggestion
-  actuator output is not supported without recompilation of the simulation firmware as posted on the :ref:`firmware server <https://firmware.ardupilot.org>` .  This is currently considered a safety feature
```

> -  parameter space is shared between the real aircraft and the simulated aircraft.  This leads to problems with sensor calibration as the sensor suite differs between the real and virtual aircraft.  Thus parameters must be wiped and rewritten when moving between real and simulating firmwares.

Reviewer: ```suggestion
-  parameter space is shared between the real aircraft and the simulated aircraft.  This leads to problems with sensor calibration as the sensor suite differs between the real and virtual aircraft.  Thus parameters must be wiped and rewritten to the vehicle when moving between real and simulation  firmware.
```

-  while it is possible to run on flash-constrained (1MB) boards, many other features will need to be compiled out to allow it to fit

> -  only quadcopters have been tested as working

Reviewer: ```suggestion
-  only Quadcopters have been tested as working
```

--------------------
Obtaining a Firmware
--------------------

Using a pre-compiled binary
...........................

> We provide a pre-compiled binary for CubeOrange at :ref:`our firmware server <https://firmware.ardupilot.org>`.

Reviewer: cant find it

Author: Fixed.
